src/v2/codegen/state.rs

```diff
@@ -325,9 +325,10 @@ pub mod util {
             return Ok(());
         }
 
-        if let Some(m) = self.infer_crate_meta()? {
+        if let m = self.infer_crate_meta()? {
```

Comment:
This doesn't seem to match any pattern. We should bind the output to some variable and then match it.

```rust
let meta = self.infer_crate_meta()?;
if let Some(m) = meta.borrow() {
    ...
}
```